benchmarks/fixed_pool_stress_test.cpp

```diff
@@ -0,0 +1,149 @@
+#include <iostream>
+#include <chrono>
+#include <string>
+#include <random>
+
+#include "umpire/config.hpp"
+
+#include "umpire/ResourceManager.hpp"
+#include "umpire/Allocator.hpp"
+#include "umpire/strategy/FixedPool.hpp"
+
+#define CONVERT 1000000 //convert sec (s) to microsec (us)
+
+const int NUM_RND = 1000; //number of rounds (used to average timing)
+const int NUM_ALLOC = 512; //number of allocations used for testing
+const int FACTOR = 2048; //Default size used when creating FixedPool (64*sizeof(int)*8)
```

Comment:
What is `FACTOR` being used for in this program?  The only time I see it being used is for `cout`.

Reply:
This could probably be done better, but it's my way of showing what the total amount of memory allocated will be. My largest size is 2^26, but it wouldn't be accurate to say 2^26 bytes is being allocated in total - it's really 2^26 * objects_per_pool (which in this case is 2048 or (64 * sizeof(int) * 8)). Factor just stores that default value for me. I've been debating modifying how I do that, but I haven't come up with anything I like better yet lol

Reply:
And please correct me if I'm wrong on that... :o

Reply:
If I understand correctly, this is stating what the `objects_per_pool` parameter for the `FixedPool` constructor is looking for:

```
  /*!
   * \brief Constructs a FixedPool.
   *
   * \param name The allocator name for reference later in ResourceManager
   * \param id The allocator id for reference later in ResourceManager
   * \param allocator Used for data allocation. It uses std::malloc
   * for internal tracking of these allocations.
   * \param object_bytes The fixed size (in bytes) for each allocation
   * \param objects_per_pool Number of objects in each sub-pool
   * internally. Performance likely improves if this is large, at
   * the cost of memory usage. This does not have to be a multiple
   * of sizeof(int)*8, but it will also likely improve performance
   * if so.
   */
  FixedPool(const std::string& name, int id, Allocator allocator,
            const std::size_t object_bytes,
            const std::size_t objects_per_pool = 64 * sizeof(int) * 8) noexcept;
```

Perhaps it would be better to specify the `objects_per_pool` directly than using the default one.

```
const std::size_t OBJECTS_PER_BLOCK{ 1 << 11 };    // 2^11 = 2048 blocks of object_bytes size
```

Reply:
Done, thanks :)